app/views/users/show.html.erb

```diff
@@ -79,11 +79,47 @@
         </table>
         </div>
         <div class="d-flex justify-content-center mx-4 mb-3 mb-lg-4">
-          <% if @user == current_user %>
-            <%= link_to "Edit profile ", edit_user_path(@user), class: "btn btn-primary btn-lg" %>
-          <% end %>
+          <%= render 'friendship_realation', object: @user%>
         </div>
       </div>
     </div>
   </div>
 </div>
+<% if current_user == @user || current_user.is_friend?(@user) %>
+  <div class="container-fluid">
+    <div class="container">
+      <div class="posts mt-5">
+        <% @posts.each do |post| %>
+          <div class="post-wrapper col-12 mb-2 p-0">
+            <div class="card">
+              <div class="card-header">
+                <%= link_to post.user.username, user_path(post.user) %><br>
+                <small class=" post-date float-right mt-1"><%= post.created_at.strftime("%Y/%m/%d") %></small>
+              </div>
+              <div class="card-body">
+                <a href="<%= post_path(post) %>" ><%= post.content %></a><br>
+                <%= like_or_dislike_btn(post) %>
+                <p>likes: <%= post.likes.count %>  comments: <%= post.comments.count %></p>
+                <% post.comments.each do |comment| %>
```

Comment:
I think this executes n+1 queries. read about `preload`, `eagerload` and `includes`